crates/sui-types/src/move_package.rs

```diff
@@ -215,6 +215,55 @@ impl MovePackage {
         )
     }
 
+    pub fn new_system(
+        version: SequenceNumber,
+        modules: Vec<CompiledModule>,
+        dependencies: impl IntoIterator<Item = ObjectID>,
+    ) -> Self {
+        let module = modules
+            .first()
+            .expect("Tried to build a Move package from an empty iterator of Compiled modules");
```

Comment:
I'm guessing these `expect`s are fine, but they make me a bit nervous... if this expect fails during an epoch change (lets say an empty vector of modules was present) would this cause issues? Or is this invariant enforced further up in the upgrade flow for system packages?

Reply:
Yes, that's right -- they are okay because we check that the modules are not empty before we sign the `ChangeEpoch` transaction (when we are doing the compatibility check, etc).